blobcache/writer.go

```diff
@@ -41,8 +41,10 @@ func (w *writer) Write(p []byte) (int, error) {
 		}
 		return 0, err
 	}
-	// w.size will be >= the actual item size since we don't use the actual
-	// amount written. Perhaps that does need to be tracked.
-	w.size += int64(n)
-	return w.w.Write(p)
+	x, err := w.w.Write(p)
+	w.size += int64(x)
+	if err != nil {
+		w.deleteOnClose = true
```

Comment:
I'm not following why this happens? Please add a code comment to explain the why of it.